src/api/meal_plan.py

```diff
@@ -1,89 +1,227 @@
+        with db.engine.begin() as connection:
+            validate_account = connection.execute(text(
+                    "SELECT id FROM accounts WHERE id = :account_id"),
+                    {"account_id": plan.account_id }
+                ).one_or_none()
 
+            if validate_account is None:
+                raise Exception("Account not found.")
 
-@router.post("/meal-plans/{meal_plan_id}/log")
-def log_meal(meal_plan_id: int, meal: MealLog):
-    with db.engine.begin() as connection:
-        connection.execute(
-            text("""
-                INSERT INTO meal_logs (meal_plan_id, date, recipe_id, quantity, meal_type)
-                VALUES (:meal_plan_id, :date, :recipe_id, :quantity, :meal_type)
-            """),
-            {
-                "meal_plan_id": meal_plan_id,
-                "date": meal.date,
-                "recipe_id": meal.recipe_id,
-                "quantity": meal.quantity,
-                "meal_type": meal.meal_type
-            }
+            result = connection.execute(
+                text("""
+                    INSERT INTO meal_plans (author_id, name, description)
+                    VALUES (:account_id, :name, :description)
+                    RETURNING id
+                """),
+                {
+                    "account_id": plan.account_id,
+                    "name": plan.name,
+                    "description": plan.description
+                }
+            )
+
+            meal_plan_id = result.scalar()
+            return {"meal_plan_id": meal_plan_id}
+    
+    except Exception as e:
+        raise HTTPException(status_code=404, detail=f"{e}")
+
+@router.delete("/{meal_plan_id}")
+def delete_meal_plan(meal_plan_id: int):
+    try:
+        with db.engine.begin() as connection:
+            validate = connection.execute(
+                text("""SELECT 1 FROM meal_plans WHERE id = :meal_plan_id"""),
+                {"meal_plan_id": meal_plan_id}
+            ).one_or_none()
+
+            if validate is None:
+                print(f"Meal Plan with ID {meal_plan_id} not found.")
+                raise Exception("Meal Plan not found.")
+
+            connection.execute(
+                text("""DELETE FROM plans_recipes WHERE meal_plan_id = :meal_plan_id"""),
+                {"meal_plan_id": meal_plan_id}
+            )
+            connection.execute(
+                text("""DELETE FROM meal_plans WHERE id = :meal_plan_id"""),
+                {"meal_plan_id": meal_plan_id}
+            )
+
+            return Response(status_code=200)
+    
+    except Exception as e:
+        raise HTTPException(
+            status_code=404, detail=f"An error occurred while deleting the meal plan: {e}"
         )
-    return {"message": "Meal logged successfully"}
 
-@router.get("/meal-plans/{meal_plan_id}/calories")
-def get_daily_calories(meal_plan_id: int, date: str):
+@router.post("/{meal_plan_id}/recipes/{recipe_id}")
+def add_recipes(meal_plan_id: int,recipe_id: int):
```

Comment:
I think this should be a POST to `/{meal_plan_id}/recipes`, with `recipe_id` included in the request JSON instead of the URL